calnex/firmware/ossfirmware.go

```diff
@@ -31,12 +33,33 @@ type OSSFW struct {
 
 // NewOSSFW returns initialized version of OSSFW
 func NewOSSFW(source string) (*OSSFW, error) {
+	var err error
+	var vs string
+	var v *version.Version
+
 	fw := &OSSFW{
 		filepath: source,
 	}
 	basename := filepath.Base(fw.filepath)
-	vs := strings.ReplaceAll(strings.TrimSuffix(basename, filepath.Ext(basename)), "sentinel_fw_v", "")
-	v, err := version.NewVersion(strings.ToLower(vs))
+	// Extract version from filename
+	// sentinel_fw_v2.13.1.0.5583D-20210924.tar
+	// calnex_combined_fw_R21.0.0.9705-20241111.tar
+
+	re, _ := regexp.Compile("(sentry_fw_|sentinel_fw_|calnex_combined_fw_)(.*).tar")
+	matches := re.FindStringSubmatch(basename)
+	if matches != nil {
+		vs = matches[2]
+	} else {
+		return nil, fmt.Errorf("invalid filename, expected prefix sentinel_fw_, sentry_fw_, or calnex_combined_fw_: %s", basename)
+	}
+
+	if strings.HasPrefix(vs, "v") {
```

Comment:
can  we somehow make this part of the regex solution above. It would still eliminate a lot of if-else-ing

Reply:
I had thought it would just make it messy but yes I think the following works. 

    re, _ := regexp.Compile("(sentry_fw_|sentinel_fw_|calnex_combined_fw_)(v[0-9].|R)(.*).tar")
    
I don't have time to test it out properly today though.